static/js/pages/applications/ApplicationDashboardPage.js

```diff
@@ -418,10 +418,20 @@ export class ApplicationDashboardPage extends React.Component<Props, State> {
     if (!application.bootcamp_run.is_payable) {
       paymentReady = false
     } else {
-      // If there are no submissions required for this application, payment should be ready after the resume
-      // requirement is fulfilled. Otherwise, it should only be ready if the last submission was approved.
-      paymentReady =
-        submissionApproved === undefined ? resumeFulfilled : submissionApproved
+      if (
+        currentUser.profile &&
+        currentUser.profile.can_skip_application_steps === true &&
+        application.bootcamp_run.allows_skipped_steps === true
```

Comment:
We should create a helper function for this in `static/js/lib/applicationApi.js`. We can call it something like `isEligibleToSkipSteps`